build-telemetry-ffi.sh

```diff
@@ -68,7 +68,7 @@ export RUSTFLAGS="${RUSTFLAGS:- -C relocation-model=pic}"
 
 datadog_telemetry_ffi="ddtelemetry-ffi"
 echo "Building the ${datadog_telemetry_ffi} crate (may take some time)..."
-cargo build --package="${datadog_telemetry_ffi}" --release --target "${target}"
+DESTDIR="$destdir" cargo build --package="${datadog_telemetry_ffi}" --release --target "${target}"
```

Comment:
I am trying to debate if this is the right way of doing things.
For now we some `cp` underneath and this different mechanism.

I don't really see the benefit from the `cp` operations bellow.

Reply:
The `OUT_DIR` set by cargo is sadly not the right place.
I think this is still slightly better, but not perfect.